lib/model.js

```diff
@@ -2708,7 +2708,7 @@ class Model {
             // Get primary keys for postgres to enable updateOnDuplicate
             options.upsertKeys = _.chain(model.primaryKeys).values().map('field').value();
             if (Object.keys(model.uniqueKeys).length > 0) {
-              options.upsertKeys = _.chain(model.uniqueKeys).values().filter(c => c.fields.length === 1).map(c => c.fields[0]).value();
+              options.upsertKeys = _.chain(model.uniqueKeys).values().filter(c => c.fields.length >= 1).map(c => c.fields).reduce(c => c[0]).value();
```

Comment:
I'm confused about the `reduce` here, would this not effectively end up returning `undefined` all the time?

cc @eseliger @sushantdhiman

Reply:
I think you are correct. The initializer for the reduce is `undefined` (not specified), and `c` is the `previous` argument, so it would effectively always return the undefined, leaving the actual values untouched.

Reply:
Due to the `.reduce` updateOnDuplicate does not work for Composite unique keys or multiple unique keys. Can you recheck for the same.